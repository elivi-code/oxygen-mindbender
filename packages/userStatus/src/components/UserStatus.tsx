import React from 'react';
import { ThemeProvider } from 'styled-components';

import { withTheme } from '@8x8/oxygen-config';
import { THEME_NAME } from '../constants/themeName';

import UserStatusIcon from './UserStatusIcon';
import UserStatusWrapper from '../styled/UserStatusWrapper';
import { UserStatusTheme } from '../types';
import { Size } from '../constants/size';
import { Status } from '../constants/status';

export interface UserStatusProps extends React.ComponentPropsWithRef<'div'> {
  theme?: Partial<UserStatusTheme>;
  size?: Size;
  status?: Status;
  title?: string;
}

const UserStatus: React.FC<UserStatusProps> = props => {
  const {
    theme, // theme is full theme object as injected by withTheme
    size = 'large',
    status = 'available',
    title,
    children,
    ...otherProps
  } = props;

  return (
    <ThemeProvider theme={theme as UserStatusTheme}>
      <UserStatusWrapper
        $size={size}
        $status={status}
        title={title}
        {...otherProps}
      >
        {children || <UserStatusIcon status={status} />}
      </UserStatusWrapper>
    </ThemeProvider>
  );
};

export default withTheme(UserStatus, THEME_NAME);
