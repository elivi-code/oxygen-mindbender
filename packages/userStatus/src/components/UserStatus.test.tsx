import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import UserStatus from './UserStatus';

jest.mock('./UserStatusIcon', () => 'UserStatusIcon');

describe('<UserStatus>', () => {
  let wrapper;

  function setup(props = {}) {
    return renderer.create(<UserStatus {...props} />).toJSON();
  }

  it('should render correctly', () => {
    wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when size is small', () => {
    wrapper = setup({ size: 'small' });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when status is away', () => {
    wrapper = setup({ status: 'away' });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when status is busy', () => {
    wrapper = setup({ status: 'busy' });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when status is doNotDisturb', () => {
    wrapper = setup({ status: 'doNotDisturb' });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when status is onBreak', () => {
    wrapper = setup({ status: 'onBreak' });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when status is onCall', () => {
    wrapper = setup({ status: 'onCall' });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when status is offline', () => {
    wrapper = setup({ status: 'offline' });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when status is workingOffline', () => {
    wrapper = setup({ status: 'workingOffline' });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when aria-label prop are provided', () => {
    wrapper = setup({
      'aria-label': 'User status aria-label',
    }) as renderer.ReactTestRendererJSON;

    expect(wrapper?.props['aria-label']).toEqual('User status aria-label');
  });
});
