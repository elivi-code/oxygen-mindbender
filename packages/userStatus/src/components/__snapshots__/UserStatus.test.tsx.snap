// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserStatus> should render correctly 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  box-sizing: border-box;
  height: 20px;
  width: 20px;
  min-width: 20px;
  min-height: 20px;
}

.c0 .presence-border {
  fill: #FFF;
}

.c0 .presence-background {
  fill: #189B55;
}

<div
  className="c0"
>
  <UserStatusIcon
    status="available"
  />
</div>
`;

exports[`<UserStatus> should render correctly when size is small 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  box-sizing: border-box;
  height: 12px;
  width: 12px;
  min-width: 12px;
  min-height: 12px;
}

.c0 .presence-border {
  fill: #FFF;
}

.c0 .presence-background {
  fill: #189B55;
}

<div
  className="c0"
>
  <UserStatusIcon
    status="available"
  />
</div>
`;

exports[`<UserStatus> should render correctly when status is away 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  box-sizing: border-box;
  height: 20px;
  width: 20px;
  min-width: 20px;
  min-height: 20px;
}

.c0 .presence-border {
  fill: #FFF;
}

.c0 .presence-background {
  fill: #DD7011;
}

<div
  className="c0"
>
  <UserStatusIcon
    status="away"
  />
</div>
`;

exports[`<UserStatus> should render correctly when status is busy 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  box-sizing: border-box;
  height: 20px;
  width: 20px;
  min-width: 20px;
  min-height: 20px;
}

.c0 .presence-border {
  fill: #FFF;
}

.c0 .presence-background {
  fill: #CB2233;
}

<div
  className="c0"
>
  <UserStatusIcon
    status="busy"
  />
</div>
`;

exports[`<UserStatus> should render correctly when status is doNotDisturb 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  box-sizing: border-box;
  height: 20px;
  width: 20px;
  min-width: 20px;
  min-height: 20px;
}

.c0 .presence-border {
  fill: #FFF;
}

.c0 .presence-background {
  fill: #CB2233;
}

<div
  className="c0"
>
  <UserStatusIcon
    status="doNotDisturb"
  />
</div>
`;

exports[`<UserStatus> should render correctly when status is offline 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  box-sizing: border-box;
  height: 20px;
  width: 20px;
  min-width: 20px;
  min-height: 20px;
}

.c0 .presence-border {
  fill: #FFF;
}

.c0 .presence-background {
  fill: #666666;
}

<div
  className="c0"
>
  <UserStatusIcon
    status="offline"
  />
</div>
`;

exports[`<UserStatus> should render correctly when status is onBreak 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  box-sizing: border-box;
  height: 20px;
  width: 20px;
  min-width: 20px;
  min-height: 20px;
}

.c0 .presence-border {
  fill: #FFF;
}

.c0 .presence-background {
  fill: #666666;
}

<div
  className="c0"
>
  <UserStatusIcon
    status="onBreak"
  />
</div>
`;

exports[`<UserStatus> should render correctly when status is onCall 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  box-sizing: border-box;
  height: 20px;
  width: 20px;
  min-width: 20px;
  min-height: 20px;
}

.c0 .presence-border {
  fill: #FFF;
}

.c0 .presence-background {
  fill: #CB2233;
}

<div
  className="c0"
>
  <UserStatusIcon
    status="onCall"
  />
</div>
`;

exports[`<UserStatus> should render correctly when status is workingOffline 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  box-sizing: border-box;
  height: 20px;
  width: 20px;
  min-width: 20px;
  min-height: 20px;
}

.c0 .presence-border {
  fill: #FFF;
}

.c0 .presence-background {
  fill: #DD7011;
}

<div
  className="c0"
>
  <UserStatusIcon
    status="workingOffline"
  />
</div>
`;
